flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/sqlserver/SQLServerSchema.java

```diff
@@ -112,25 +214,80 @@ protected void doClean() throws SQLException {
     }
 
     /**
+     * Query objects with any of the given types.
+     *
+     * @param types the object types to be queried
+     * @return the found objects
+     * @throws SQLException when the retrieval failed
+     */
+    private List<DBObject> queryDBObjects(ObjectType... types) throws SQLException {
```

Comment:
Instead of relying on several tables of the information_schema, the proposed code makes only use of the sys.objects table whenever possible. This make joins based on object ids (required to query extended property where the Microsoft tools flags are stored) very straightforward.

However, this imply that we also must select a bit more precisely the objects we are interested only (not only function and procedure, but low-level types such as "SQL function", "CLR function", "SQL Procedure", "CLR procedure, etc."). Thus, the additional testing required and implemented in the two first commit of this PR.